src/GraphQL.Tests/AuthorizationTests.cs

```diff
@@ -0,0 +1,116 @@
+using GraphQL.Types;
+
+namespace GraphQL.Tests;
+
+public class AuthorizationTests
+{
+    [Fact]
+    public void Field()
+    {
+        var field = new FieldType();
+        field.RequiresAuthorization().ShouldBeFalse();
+        field.AuthorizeWith("Policy1");
+        field.RequiresAuthorization().ShouldBeTrue();
+        field.AuthorizeWith("Policy2");
+        field.AuthorizeWith("Policy2");
+        field.AuthorizeWithRoles("Role1,Role2");
+        field.AuthorizeWithRoles("Role3, Role2");
+
+        field.RequiresAuthorization().ShouldBeTrue();
+        field.GetPolicies().ShouldBe(new string[] { "Policy1", "Policy2" });
+        field.GetRoles().ShouldBe(new string[] { "Role1", "Role2", "Role3" });
+    }
+
+    [Fact]
+    public void FieldBuilder()
+    {
+        var graph = new ObjectGraphType();
+        graph.Field<StringGraphType>("Field")
+            .AuthorizeWith("Policy1")
+            .AuthorizeWith("Policy2")
+            .AuthorizeWith("Policy2")
+            .AuthorizeWithRoles("Role1,Role2")
+            .AuthorizeWithRoles("Role3, Role2");
+
+        var field = graph.Fields.Find("Field");
+        field.RequiresAuthorization().ShouldBeTrue();
+        field.GetPolicies().ShouldBe(new string[] { "Policy1", "Policy2" });
+        field.GetRoles().ShouldBe(new string[] { "Role1", "Role2", "Role3" });
+    }
+
+    [Fact]
+    public void ConnectionBuilder()
+    {
+        var graph = new ObjectGraphType();
+        graph.Connection<StringGraphType>()
+            .Name("Field")
+            .AuthorizeWith("Policy1")
+            .AuthorizeWith("Policy2")
+            .AuthorizeWith("Policy2")
+            .AuthorizeWithRoles("Role1,Role2")
+            .AuthorizeWithRoles("Role3, Role2");
+
+        var field = graph.Fields.Find("Field");
+        field.RequiresAuthorization().ShouldBeTrue();
+        field.GetPolicies().ShouldBe(new string[] { "Policy1", "Policy2" });
+        field.GetRoles().ShouldBe(new string[] { "Role1", "Role2", "Role3" });
+    }
+
+    [Fact]
+    public void AutoOutputGraphType()
```

Comment:
This test demonstrates and tests the manner in which the `GraphQLAuthorizeAttribute` can be used without schema-first.